variant-remapping-automation/get_custom_assembly.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python
+
+# Copyright 2019 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import os
+import re
+import shutil
+import sys
+import urllib
+from copy import copy
+from csv import DictReader, excel_tab, DictWriter
+
+from cached_property import cached_property
+from ebi_eva_common_pyutils.config import cfg
+from ebi_eva_common_pyutils.logger import AppLogger, logging_config
+from ebi_eva_common_pyutils.metadata_utils import get_metadata_connection_handle
+from ebi_eva_common_pyutils.pg_utils import get_all_results_for_query
+from retry import retry
+
+sys.path.append(os.path.dirname(__file__))
+
+from remapping_config import load_config
+
+
+class CustomAssembly(AppLogger):
+    """
+    This class creates a custom assembly based on existing assembly fasta and report.
+    It adds additional contigs provided by the required_contigs function after checking that they are not already in
+    the assembly.
+    """
+    def __init__(self, assembly_accession, assembly_fasta_path, assembly_report_path, eutils_api_key=None):
+        self.assembly_accession = assembly_accession
+        self.assembly_fasta_path = assembly_fasta_path
+        self.assembly_report_path = assembly_report_path
+        self.eutils_api_key = eutils_api_key or cfg.get('eutils_api_key')
+        self.assembly_report_headers = None
+
+    @property
+    def output_assembly_report_path(self):
+        base, ext = os.path.splitext(self.assembly_report_path)
+        return base + '_custom' + ext
+
+    @property
+    def output_assembly_fasta_path(self):
+        base, ext = os.path.splitext(self.assembly_fasta_path)
+        return base + '_custom' + ext
+
+    @property
+    def assembly_directory(self):
+        return os.path.dirname(self.assembly_fasta_path)
+
+    @cached_property
+    def required_contigs(self):
+        raise NotImplementedError
```

Comment:
Could you add a docstring for what this property needs to look like, i.e. a `list` of `dict` with the following keys...

Reply:
Done